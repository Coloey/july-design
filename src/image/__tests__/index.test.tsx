import React from "react"
import { fireEvent, render,screen} from '@testing-library/react';
import Image from "../index";
import "@testing-library/jest-dom"
describe("Image[Base]",() => {
   it("should render with src",() => {
        const {asFragment} = render(<Image src='https://avatars.githubusercontent.com/u/71813516?v=4'></Image>)
        expect(asFragment().firstChild).toMatchSnapshot();
    })
   /* it("className and style props should work on img element",() => {
        const onClick = jest.fn()
        render(
            <Image src='https://avatars.githubusercontent.com/u/71813516?v=4'
            onClick={onClick}
            className="img"
            style={{
                objectFit:'cover'
            }}
            ></Image>
        ) 
        const img = screen.getByTestId("img");
        expect(img).toHaveClass('img')
        expect(img).toHaveStyle({ objectFit:'cover' })
    })
   /* it("with click when disable preview",() => {

    })*/
    
})


