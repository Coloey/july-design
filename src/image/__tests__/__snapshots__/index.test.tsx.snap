// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image[Base] should render with src 1`] = `null`;
